keps/sig-node/1985-remove-dockershim/README.md

```diff
@@ -0,0 +1,298 @@
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Terms
+
+- **CRI:** Container Runtime Interface – a plugin interface which enables kubelet to use a wide variety of container 
+runtimes, without the need to recompile.
+
+## Summary
+
+CRI for docker (i.e. dockershim) is currently a built-in container runtime in kubelet code base. This proposal aims 
+at a deprecation and subsequent removal of dockershim from kubelet.
+
+## Motivation
+
+In Kubernetes, the CRI interface is used to talk to a container runtime, The design of CRI is to be able to run a CRI
+implementation as a separate binary. However currently the CRI of docker (a.k.a. dockershim) is part of kubelet code, runs
+as part of kubelet and is tightly coupled with kubelet's lifecycle. 
+
+This is not ideal as kubelet then has dependency on specific container runtime which leads to maintenance burden for not 
+only developers in sig-node, but also cluster administrators when critical issues (e.g. runc CVE) happen to container 
+runtimes. The pros of removing dockershim is straightforward:
+
+### Pros
+- Docker is not special and should be just a CRI implementation just like every other CRI implementation in our ecosystem. 
+- Currently, dockershim "enjoys" some inconsistent integrations for various reasons (see [legacyLogProvider](https://cs.k8s.io/?q=legacyLogProvider&i=nope&files=&repos=kubernetes/kubernetes) for example) . Removing these "features" should eliminate maintenance burden of kubelet.
+- A cri-dockerd can be maintained independently by folks who are interested in keeping this functionality
+- Over time we can remove vendored docker dependencies in kubelet.
```

Comment:
Another two pros in my mind:

1) Due to convenience inheriting from this builtin shim for the container runtime, there is less incentive to move to new container runtime. The production issues found and addressed in both Containerd and CRI-O might cause the production issues for some users, which causes the maintenance burden to Kubernetes community.
2) The community can focus on the new container runtimes enhancements.

Reply:
Included these 2 suggestions. thanks @dchen1107